app-text/llpp/llpp-31.ebuild

```diff
@@ -0,0 +1,72 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs xdg-utils
+
+DESCRIPTION="Graphical PDF viewer which aims to superficially resemble less(1)"
+HOMEPAGE="https://github.com/moosotc/llpp"
+SRC_URI="https://github.com/moosotc/llpp/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="public-domain"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="+ocamlopt static-libs"
+
+RDEPEND=">=app-text/mupdf-1.15.0:0=[static-libs=]
+	media-libs/openjpeg:2[static-libs=]
+	media-libs/fontconfig:1.0[static-libs=]
+	media-libs/freetype:2[static-libs=]
+	media-libs/jbig2dec:=[static-libs=]
+	sys-libs/zlib[static-libs=]
+	virtual/jpeg:0[static-libs=]
+	x11-libs/libX11[static-libs=]
+	x11-misc/xsel"
+DEPEND="${RDEPEND}
+	app-text/asciidoc
+	virtual/pkgconfig
+	static-libs? (
+		app-arch/bzip2[static-libs]
```

Comment:
I'm confused here. These are somehow only dependencies if you're statically linking? I think you may want [static-libs?] instead, but also, we try to chuck out the static-libs USE flag where it's pointless.

Reply:
That's the way those dependencies have been split since before Gentoo migrated from cvs to git in commit 56bd759df1d0 five years ago. It seems llpp-15 was the first version to do so. Let's see the historical archives...

Initially in llpp-15.ebuild [v1.4](https://gitweb.gentoo.org/repo/gentoo/historical.git/commit/app-text/llpp/llpp-15.ebuild?id=3b79715b4d67849655b0e69542b45168f2711ded) the list in RDEPEND and DEPEND had the same packages. This was in 2013-06-08.

[v1.6](https://gitweb.gentoo.org/repo/gentoo/historical.git/commit/app-text/llpp/llpp-15.ebuild?id=5cda2b514ac5d2311398a24c2914dfe381067a70) added those five packages in DEPEND but not in RDEPEND with a commit message "Improve static build" in 2013-06-13.

And that's all I know.

Looking at the linkage of the non-static binary, I do believe that three of those five should move to RDEPEND with `[static-libs=]` since they are getting linked anyway by some other dependency that may or may not enforce matching statics but x11-libs/libXmu and media-libs/libXcm probably should not even be there anymore. I don't see anything referring to them.

Doing that.

Changing `[static-libs=]` to `[static-libs?]` feels correct, doing so for all those deps. The ebuild itself originally had `IUSE="static"` instead but that was changed in [PR 13640](https://github.com/gentoo/gentoo/pull/13640).